test_rclcpp/test/test_subscription.cpp

```diff
@@ -24,12 +25,44 @@
 #ifdef RMW_IMPLEMENTATION
 # define CLASSNAME_(NAME, SUFFIX) NAME ## __ ## SUFFIX
 # define CLASSNAME(NAME, SUFFIX) CLASSNAME_(NAME, SUFFIX)
+
 #else
 # define CLASSNAME(NAME, SUFFIX) NAME
 #endif
 
+#define xstr(s) str(s)
+#define str(s) #s
+
 static const std::chrono::milliseconds sleep_per_loop(10);
-static const int max_loops = 200;
+static const size_t max_loops = 200;
+
+// Sleep for timeout ms or until a subscriber has registered for the topic
+void busy_wait_for_subscriber(
+  std::shared_ptr<const rclcpp::Node> node,
+  const std::string & topic_name,
+  std::chrono::milliseconds timeout = std::chrono::milliseconds(1),
+  std::chrono::microseconds sleep_period = std::chrono::microseconds(100))
+{
+  std::chrono::microseconds time_slept(0);
+#ifdef RMW_IMPLEMENTATION
+  if (strcmp(xstr(RMW_IMPLEMENTATION), "rmw_fastrtps_cpp") == 0) {
```

Comment:
no, unfortunately, two levels of macros is necessary. `#RMW_IMPLEMENTATION` does not compile, and with only one level macro the test does not pass.

https://gcc.gnu.org/onlinedocs/cpp/Stringification.html